File: 단기43차/Java(eclipse)/0522/src/Bank.java
import java.util.Scanner;

//Bank 가 Wbarry를 소유하고 있다. (has a 3번째패턴, 생성주기를 동일하게)
public class Bank {
	private Wbarray acclist;
	private Scanner scan = new Scanner(System.in);
	
	public Bank() {
		acclist = new Wbarray(10);
	}
	
	public void Makeaccount() {
		
		try {
		System.out.println("이    름 : ");
		String name = scan.nextLine();
		
		System.out.print(" 입 금 액 : ");
		int money = Integer.parseInt(scan.nextLine());
		
		Account acc = new Account(name, money);
		
		acclist.push_back(acc);
		System.out.println("계좌가 저장되었습니다.");
	} catch (Exception e) {
		System.out.println("저장 에러 : " + e.getMessage());
	}	
	}
	//select
	public void Printallaccount() {
		for (int i = 0; i <acclist.getSize(); i++) {
			Account acc = acclist.getAccount(i);
			acc.printline();
		}
		
	}
	catch(Exception e) {
		System.out.println("전체출력에러 : " + e.getMessage());
	}
	public void selectaccount() {
		System.out.print("검색할 이름 : ");
		String name = scan.nextLine();
		
		int idx = NameToIdx(name);
		Account acc = acclist.getAccount(idx);
		acc.print();
	}
	catch(Exception e) {
		System.out.println("검색 에러" + e.getMessage());
	}
	public void Addmoney() {
		System.out.print("입금할 계좌 고객명 : ");
		String name = scan.nextLine();
		
		int idx = NameToIdx(name);
		Account acc = acclist.getAccount(idx);
		acc.print();
	
		System.out.print("입금액 : ");
	int money = Integer.parseInt(scan.nextline());
	acc.inputmoney(money);

	System.out.print("입금되었슴니다.");
	
	}
	catch(Exception e) {
		System.out.println("검색 에러" + e.getMessage());
	}
	public void Minmoney() {
		
	}
	public void DeleteAccount() {
		
	}
	private int NameToIdx (String name) {
		
		for (int i = 0; i < acclist.getSize(); i++) {
			Account acc = acclist.getAccount(i);
			if(acc.getName())
		}
	}
}

File: 단기43차/C언어+Cpp/참고(짝꿍)/C 언어/0417/0417/bank.cpp
//bank.cpp
#include <iostream>
using namespace std;
#include "account.h"
#include "wbarray.h"
#include "bank.h"



bank::bank()
{
	acclist = new wbarray;
}


bank::~bank()
{
	delete acclist;
}


void bank::makeaccount()		//insert
{
	try
	{
		//1. 필요한 정보를 입력 받는다.
		int accid, balance;
		char name[20], date[20];

		cout << "이름 : ";	cin >> name;
		cout << "계좌번호 : ";	cin >> accid;
		cout << "입금액 : ";	cin >> balance;
		cout << "개설일 : ";	cin >> date;

		//2. 저장 객체 생성한다.
		account *acc = new account(name, accid, balance, date);

		//3. wbarray에 저장 요청한다.
		acclist->push_back(acc);

		cout << "저장되었습니다." << endl;
	}
	catch (const char* msg)
	{
		cout << msg << endl;
	}
}


void bank::printallaccount()
{
	//순회
	for (int i = 0; i < acclist->getsize(); i++)
	{
		account *acc = (account*)acclist->getdata(i);
		acc->printline();
	}
}


void bank::selectaccount()
{
	try
	{
		//1. 검색키 얻기
		char name[20];
		cout << "검색할 이름 입력 : ";	cin >> name;

		//2. 검색 알고리즘
		int idx = nametoidx(name);

		//3. 결과 출력
		account *acc = (account*)acclist->getdata(idx);
		acc->print();
	}
	catch (const char* msg)
	{
		cout << msg << endl;
	}
}


void bank::addmoney()		//update
{
	try
	{
		//1. 검색키 얻기
		char name[20];
		cout << "검색할 이름 입력 : ";	cin >> name;

		//2. 검색 알고리즘
		int idx = nametoidx(name);

		//3. 결과 출력
		account *acc = (account*)acclist->getdata(idx);
		acc->print();

		int money;
		cout << "입금액 : ";		cin >> money;
		acc->inputmoney(money);
	}
	catch (const char* msg)
	{
		cout << msg << endl;
	}
}


void bank::minmoney()		//update
{
	try
	{
		//1. 검색키 얻기
		char name[20];
		cout << "검색할 이름 입력 : ";	cin >> name;

		//2. 검색 알고리즘
		int idx = nametoidx(name);

		//3. 결과 출력
		account *acc = (account*)acclist->getdata(idx);
		acc->print();

		int money;
		cout << "출금액 : ";		cin >> money;
		acc->outputmoney(money);
	}
	catch (const char* msg)
	{
		cout << msg << endl;
	}
}


void bank::deleteaccount()
{
	try
	{
		//1. 삭제키 얻기
		char name[20];
		cout << "삭제할 이름 입력 : ";	cin >> name;

		//2. 검색 알고리즘
		int idx = nametoidx(name);

		//3. 결과 출력
		acclist->erase(idx);

		cout << "삭제되었습니다." << endl;
	}
	catch (const char* msg)
	{
		cout << msg << endl;
	}
}


int bank::nametoidx(string name)
{
	for (int i = 0; i < acclist->getsize(); i++)
	{
		account *acc = (account*)acclist->getdata(i);
		if (acc->getname() == name)
			return i;
	}
	throw "데이터가 존재하지 않습니다.";
}
